components/service-operator/apis/access/v1beta1/principal_types_test.go

```diff
@@ -0,0 +1,78 @@
+package v1beta1_test
+
+import (
+	"os"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/alphagov/gsp/components/service-operator/apis/access/v1beta1"
+	"github.com/alphagov/gsp/components/service-operator/internal/aws/cloudformation"
+)
+
+var _ = Describe("Principal", func() {
+
+	var o v1beta1.Principal
+
+	BeforeEach(func() {
+		os.Setenv("CLUSTER_NAME", "xxx") // required for env package
+		o = v1beta1.Principal{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "example",
+				Namespace: "default",
+				Labels: map[string]string{
+					cloudformation.AccessGroupLabel: "test.access.group",
+				},
+			},
+		}
+	})
+
+	It("should return a unique role name", func() {
+		Expect(o.GetRoleName()).To(Equal("svcop-xxx-default-example"))
+	})
+
+	It("should implement runtime.Object", func() {
+		o2 := o.DeepCopyObject()
+		Expect(o2).ToNot(BeZero())
+	})
+
+	Context("cloudformation", func() {
+
+		It("should generate a unique stack name prefixed with cluster name", func() {
+			Expect(o.GetStackName()).To(HavePrefix("xxx-principal-default-example"))
+		})
+
+		It("should have outputs for role name", func() {
+			t := o.GetStackTemplate()
+			Expect(t.Outputs).To(HaveKey("IAMRoleName"))
+		})
+
+		It("should whitelist the IAMRoleName output", func() {
```

Comment:
Safelist instead of whitelist?

Reply:
I've never heard the term "safelist" outside of autom8 ... What is the reason to go against the popular whitelisting/blacklisting terms?

Reply:
While the terms whitelist and blacklist are currently more common, and I'm not asking for all the existing usage to be replaced, it would be good to start using the more descriptive and less evocative safelist / blocklist / denylist where possible as we update and add to our docs and code. We made a similar choice to stop using master and slave in a number of our Jenkins repos and I see this as a similar use case.

Reply:
interesting, thanks @deanwilson @issyl0

Reply:
_corrected_